bindings/trycp_client/src/lib.rs

```diff
@@ -0,0 +1,888 @@
+use holochain_client::AgentSigner;
+use holochain_conductor_api::ZomeCall;
+use holochain_types::prelude::{ExternIO, FunctionName, ZomeName};
+use holochain_zome_types::cell::CellId;
+use holochain_zome_types::prelude::ZomeCallUnsigned;
+use serde::de::DeserializeOwned;
+use std::io;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio_tungstenite::tungstenite::client::IntoClientRequest;
+use trycp_api::MessageResponse;
+use trycp_client::{Request, SignalRecv, TrycpClient};
+use wind_tunnel_instruments::{OperationRecord, Reporter};
+use wind_tunnel_instruments_derive::wind_tunnel_instrument;
+
+pub mod prelude {
+    pub use super::TryCPClientInstrumented as TryCPClient;
+}
+
+#[derive(Clone)]
+pub struct TryCPClientInstrumented {
+    trycp_client: Arc<TrycpClient>,
+    signal_recv: Arc<tokio::sync::Mutex<SignalRecv>>,
+    signer: Arc<dyn AgentSigner + Send + Sync>,
+    reporter: Arc<Reporter>,
+    timeout: Duration,
+}
+
+impl std::fmt::Debug for TryCPClientInstrumented {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TryCPClient").finish()
+    }
+}
+
+mod control_impl {
+    use super::*;
+    use trycp_client::Signal;
+
+    impl TryCPClientInstrumented {
+        pub async fn connect<R>(
+            request: R,
+            signer: Arc<dyn AgentSigner + Send + Sync>,
+            reporter: Arc<Reporter>,
+        ) -> io::Result<Self>
+        where
+            R: IntoClientRequest + Unpin,
+        {
+            let (trycp_client, signal_recv) = TrycpClient::connect(request).await?;
+            Ok(Self {
+                trycp_client: Arc::new(trycp_client),
+                signal_recv: Arc::new(tokio::sync::Mutex::new(signal_recv)),
+                signer,
+                reporter,
+                timeout: Duration::from_secs(30),
+            })
+        }
+
+        pub async fn recv_signal(&mut self) -> Option<Signal> {
+            let mut recv = self.signal_recv.lock().await;
+            recv.recv().await
+        }
```

Comment:
```suggestion
        pub async fn recv_signal(&self) -> Option<Signal> {
            let mut recv = self.signal_recv.lock().await;
            recv.recv().await
        }
```

No longer needs to be `mut` if wrapped in a mutex.

Reply:
Good spot, outcompeting clippy there :)